### PR TITLE
Update "x" icon in Dialog

### PR DESCRIPTION
**Pull request by LMS007**
- Replace the ad-hoc "x" with an IconButton
- Set the line-height of the title to 1em which makes it a bit taller only if there is no cancel "x" icon rendered.
- The cancel icon is copied from the client and eventually could move to the shared package.

To test, use this assignment, the URLpicker or File picker both use this dialog.
https://hypothesis.instructure.com/courses/125/assignments/1257

**Before**
Screen Shot 2021-05-03 at 9 32 08 AM

**After**
Screen Shot 2021-05-03 at 9 31 13 AM

The header is slightly taller because the negative margin has been removed from the button. I think we can adjust the margin a bit more but perhaps with the dialog updates to come?

### CODE REVIEW

File: lms/static/styles/components/_Dialog.scss [hunk 1]
```diff
@@ -40,26 +40,15 @@
...
-  // Add negative margins so that the button does not force the dialog to
-  // grow in height.
-  font-size: var.$title-font-size;
-  padding: 5px;
-  margin: -10px 0px;
-
-  &:hover {
-    cursor: pointer;
-    color: var.$brand;
+.Dialog__cancel-icon-btn {
```
Comment (lyzadanger) [inline lms/static/styles/components/_Dialog.scss:47]: Please isolate/consolidate all customizations of the shared button styling into a single module, `buttons.scss`. See the client for how this is done there.

**Review by lyzadanger — CHANGES_REQUESTED** [review]
Hi! Can you please extract any customizations of the shared-button styling into a single module (see the client for how that is done there?). I'm trying to make sure we consolidate any needed customizations and don't have dependencies scattered around the app. Thanks!

File: lms/static/styles/components/_Dialog.scss (hunk 1 above)
Comment (lyzadanger) [reply lms/static/styles/components/_Dialog.scss:47]: Actually, wait. Can you walk me through what this customization is trying to achieve?

Comment (LMS007) [reply lms/static/styles/components/_Dialog.scss]: Thanks! that seems reasonable. In the future, I'll make a note to add the buttons.scss file which we will probably need to get into with the grader.

**Review by lyzadanger — APPROVED** [review]
Cool. Thanks!

**Comment by robertknight** [comment]
@esanzgar noticed a regression caused by this PR in [the file picker dialog](https://github.com/hypothesis/lms/issues/2667#issuecomment-836571230). In the immediate term I'm going to [revert this PR](https://github.com/hypothesis/lms/pull/2688) to resolve the issue and we can revisit later. @LMS007 can you confirm if you are able to reproduce the issue on your system (using a checkout of master prior to the merge of https://github.com/hypothesis/lms/pull/2688)? The PR description here mentions that the file picker dialog uses the `Dialog` component so I'm wondering how it got overlooked.

**Pull request merged**